File: modern-app/src/main/java/demo/routing/SessionStoringController.java
package demo.routing;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.http.HttpSession;

@RestController
@RequestMapping("/routes")
public class SessionStoringController {

    @Autowired
    SessionDemo sessionDemo;

    @GetMapping("/withsession")
    public String store(HttpSession httpSession) {
        httpSession.setAttribute("from", "pcf");
        return "session-from-pcf: counter: " + sessionDemo.getCounter();
    }

    @GetMapping("/logoff")
    public void logoff(HttpSession httpSession){
        httpSession.invalidate();
    }
}

File: modern-app/src/main/kotlin/demo/routing/SessionStoringController.kt
package demo.routing

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

import javax.servlet.http.HttpSession

@RestController
@RequestMapping("/routes")
class SessionStoringController {

    @Autowired
    private var sessionDemo: SessionDemo? = null

    @GetMapping("/withsession")
    fun store(httpSession: HttpSession): String {
        httpSession.setAttribute("from", "pcf")
        return "session-from-pcf: counter: " + sessionDemo?.getCounter()
    }

    @GetMapping("/logoff")
    fun logoff(httpSession: HttpSession) {
        httpSession.invalidate()
    }
}
